format/tap/tap.go

```diff
@@ -0,0 +1,146 @@
+package tzx
+
+// https://worldofspectrum.net/zx-modules/fileformats/tapformat.html
+
+import (
+	"embed"
+
+	"golang.org/x/text/encoding/charmap"
+
+	"github.com/wader/fq/format"
+	"github.com/wader/fq/pkg/decode"
+	"github.com/wader/fq/pkg/interp"
+	"github.com/wader/fq/pkg/scalar"
+)
+
+//go:embed tap.md
+var tapFS embed.FS
+
+func init() {
+	interp.RegisterFormat(
+		format.TAP,
+		&decode.Format{
+			Description: "TAP tape format for ZX Spectrum computers",
+			DecodeFn:    tapDecoder,
```

Comment:
Add this and i think the `read_one_block` option should work:
```go
			DefaultInArg: format.TAP_In{
				ReadOneBlock: false,
			},
```
It's how the fq internals know the type and also default values

Reply:
I did initially think to add this, but then figured, it's false by default so no need to add it :)

Reply:
Aha :) i've thought about making `Format` generic somehow then the in/out types could be passed as use zero value... but maybe not worth it? maybe will refactor this in the future